tests/test_devices.py

```diff
@@ -255,6 +255,152 @@ def test_device_inventory_update_success(client, mocker):
     assert device._inventory == inventory
 
 
+class test_inputs:
+    def test_property_populated(client, mocker):
+        """Should check if the device property are correctly populated"""
+        device = AlarmDevice(client)
+        mocker.spy(device._connection, "query")
+        device.connect("username", "password")
+        inputs = {
+            0: "Entryway Sensor",
+            1: "Outdoor Sensor 1",
+            2: "Outdoor Sensor 2",
+        }
+        # Test
+        device.update()
+        device_inputs = {}
+        for input_id, name in device.inputs:
```

Comment:
`device.inputs` is a generator, so you can't compare directly it with `inputs`. Generators must be iterated so I understand your need of `device_inputs` and the `for` cycle! That said, you can use some Python syntactic sugar to make what you are doing more concise!

You can use the `dict()` constructor (standard library) that will automatically do everything for you! In this specific case, you can simply:
```python
        inputs = {
            0: "Entryway Sensor",
            1: "Outdoor Sensor 1",
            2: "Outdoor Sensor 2",
        }
        # Test
        device.update()
        assert inputs == dict(device.inputs)
```

And that's it! Do the same for all the tests, assuming what you return is always a `dict` object.